x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForSnapshotStep.java

```diff
@@ -104,7 +104,7 @@ public void evaluateCondition(Metadata metadata, Index index, Listener listener,
         );
         String snapshotName = snapPolicyMeta.getLastSuccess().getSnapshotName();
         String repositoryName = snapPolicyMeta.getPolicy().getRepository();
-        GetSnapshotsRequest request = new GetSnapshotsRequest().repositories(repositoryName)
+        GetSnapshotsRequest request = new GetSnapshotsRequest(TimeValue.MAX_VALUE).repositories(repositoryName)
```

Comment:
This timeout was missing previously (i.e. defaulted to 30s) but like other ILM actions it should have been `MAX_VALUE`.